pkg/function/server/server_test.go

```diff
@@ -78,20 +79,71 @@ func Test_server_map(t *testing.T) {
 	}
 }
 
-func Test_server_reduce(t *testing.T) {
+func Test_server_mapT(t *testing.T) {
 	file, err := os.CreateTemp("/tmp", "numaflow-test.sock")
 	assert.NoError(t, err)
 	defer func() {
 		err = os.RemoveAll(file.Name())
 		assert.NoError(t, err)
 	}()
+	tests := []struct {
+		name   string
+		fields fields
+	}{
+		{
+			name: "server_mapT",
+			fields: fields{
+				mapTHandler: functionsdk.MapTFunc(func(ctx context.Context, key string, d functionsdk.Datum) functionsdk.MessageTs {
+					msg := d.Value()
+					return functionsdk.MessageTsBuilder().Append(functionsdk.MessageTTo(time.Time{}, key+"_test", msg))
+				}),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// note: using actual UDS connection
 
-	var testKey = "reduce_key"
+			go New().RegisterMapperT(tt.fields.mapTHandler).Start(context.Background(), WithSockAddr(file.Name()))
```

Comment:
will this go-routine be leaked after the test is over?

Reply:
Good catch. Will fix. Thanks!